jobs/apply-specs/templates/bin/deploy-specs.erb

```diff
@@ -17,63 +17,39 @@ wait_for() {
     ${kubectl} rollout status "deployments/${1}" -w --namespace=kube-system
 }
 
-create_heapster_secrets() {
-  local secret_name=heapster-certs
-  local namespace=kube-system
-
-  if ${kubectl} get secret ${secret_name} -n ${namespace}  > /dev/null 2>&1; then
-    ${kubectl} delete secret ${secret_name} -n ${namespace}
+delete_heapster_if_it_exists() {
+  if ${kubectl} -n kube-system get service/heapster > /dev/null 2>&1; then
+    ${kubectl} delete -f ${spec_dir}/heapster/heapster.yml
   fi
-
-  ${kubectl} create secret generic ${secret_name} \
-  --from-file=heapster-cert=${config_dir}/heapster/heapster.crt \
-  --from-file=heapster-key=${config_dir}/heapster/heapster.key \
-  --from-file=ca=${config_dir}/ca.pem \
-  -n ${namespace}
 }
 
-create_influx_secrets() {
-  local secret_name=influxdb-certs
-  local namespace=kube-system
-
-  if ${kubectl} get secret ${secret_name} -n ${namespace} > /dev/null 2>&1; then
-    ${kubectl} delete secret ${secret_name} -n ${namespace}
+delete_influxdb_if_it_exists() {
+  if ${kubectl} -n kube-system get service/monitoring-influxdb > /dev/null 2>&1; then
+    ${kubectl} delete -f ${spec_dir}/influxdb/influxdb.yml
   fi
-
-  ${kubectl} create secret generic ${secret_name} \
-  --from-file=influxdb-cert=${config_dir}/influxdb/influxdb.crt \
-  --from-file=influxdb-key=${config_dir}/influxdb/influxdb.key \
-  -n ${namespace}
-}
-
-create_dashboard_secrets() {
-  local secret_name=kubernetes-dashboard-certs
-  local namespace=kube-system
-
-  if ${kubectl} get secret ${secret_name} -n ${namespace} > /dev/null 2>&1; then
-    ${kubectl} delete secret ${secret_name} -n ${namespace}
-  fi
-
-  ${kubectl} create secret generic ${secret_name} \
-  --from-file=kubernetes-dashboard-cert=${config_dir}/kubernetes-dashboard/kubernetes-dashboard.crt \
-  --from-file=kubernetes-dashboard-key=${config_dir}/kubernetes-dashboard/kubernetes-dashboard.key \
-  -n ${namespace}
 }
 
 main() {
-  apply_spec "kube-dns.yml"
+  <%- if p('delete-heapster') then -%>
+  delete_heapster_if_it_exists
+  delete_influxdb_if_it_exists
+  <%- end -%>
+
+  apply_spec "kube-dns/kube-dns.yml"
   wait_for kube-dns
 
-  create_heapster_secrets
-  apply_spec "heapster.yml"
+  <%- if !p('delete-heapster') then -%>
+  apply_spec "heapster/heapster.yml"
   wait_for heapster
 
-  create_influx_secrets
-  apply_spec "influxdb.yml"
+  apply_spec "influxdb/influxdb.yml"
   wait_for monitoring-influxdb
+  <%- end -%>
+
+  apply_spec "metrics-server/"
+  wait_for metrics-server
 
-  create_dashboard_secrets
-  apply_spec "kubernetes-dashboard.yml"
+  apply_spec "kubernetes-dashboard/kubernetes-dashboard.yml"
   wait_for kubernetes-dashboard
 
   <% if_link('cloud-provider') do |cloud_provider| %>
```

Comment:
To apply the addons we need to change the path.
Should be 
```
 apply_spec "addon-spec/addon-spec.yml"
```
Instead of 
```
 apply_spec "addon-spec.yml"
```